tests/integration/kubernetes/k8s-confidential.bats

```diff
@@ -10,17 +10,22 @@ load "${BATS_TEST_DIRNAME}/confidential_common.sh"
 load "${BATS_TEST_DIRNAME}/tests_common.sh"
 
 setup() {
-	SUPPORTED_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+	SUPPORTED_TEE_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+	SUPPORTED_NON_TEE_HYPERVISORS=("qemu")
 
 	# This check must be done with "<SPACE>${KATA_HYPERVISOR}<SPACE>" to avoid
 	# having substrings, like qemu, being matched with qemu-$something.
-	[[ " ${SUPPORTED_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]] ||  skip "Test not supported for ${KATA_HYPERVISOR}."
+	if [[ " ${SUPPORTED_TEE_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]] || [[ " ${SUPPORTED_NON_TEE_HYPERVISORS[*]} " =~  " ${KATA_HYPERVISOR} " ]]; then
```

Comment:
I think this check is wrong.  Reading it it says:
* If kata-hypervisor is in the supported list of TEEs, or if kata-hypervisor is in the supported list of non-TEEs
  * then skip the test

What we would actually want to check is:
* If kata-hypervisor is **NOT** in the supported list of tees **AND** kata-hypervisor is **NOT** in the supported list of non-tees
  *  then skip the test.